core/src/main/java/com/google/errorprone/bugpatterns/MemberName.java

```diff
@@ -143,7 +145,10 @@ private static boolean hasTestAnnotation(MethodSymbol symbol) {
   public Description matchVariable(VariableTree tree, VisitorState state) {
     VarSymbol symbol = getSymbol(tree);
     String name = tree.getName().toString();
-    if (symbol.owner instanceof MethodSymbol && symbol.getKind().equals(ElementKind.PARAMETER)) {
+    if (symbol.owner instanceof MethodSymbol
+        && symbol.getKind() == ElementKind.PARAMETER
+        && ASTHelpers.findEnclosingNode(state.getPath(), Tree.class).getKind()
+            != Kind.LAMBDA_EXPRESSION) {
```

Comment:
Initially I wrote a fix that tests for `index < 0` below, but then malformed lambda parameter names aren't flagged. I updated the test accordingly. This does make me wonder: are there possibly other expression types that should be excluded? (Nothing comes to mind, but.)